### PR TITLE
Dev checkout

### PR DESCRIPTION
**Pull request by archit7448**
###  What I did : 

- Add coupon 
- refactor the code 

 ### files to check 
- cart page.jsx 
- reducer function

### CODE REVIEW

**Review by DivyanshSareen — APPROVED** [review]
Clean code, the UI is easy to navigate. Cool feedback alerts.

**Review by srishtigautam21 — APPROVED** [review]
Clean understandable code
Good to merge

File: src/pages/cartPage/cartPage.jsx [hunk 1]
```diff
@@ -2,15 +2,49 @@ import { Header } from "../../Components/header/header";
...
+  const [error, setError] = useState(false);
+  const [coupon, setCoupon] = useState("");
+  const [show, setShow] = useState(false);
   const TotalPrice = () => {
     const totalPrice = cart.map(
       ({ price, quantity }) => Number(price) * Number(quantity)
     );
     return totalPrice.reduce((acc, curr) => acc + curr, 0);
   };
+  console.log(TotalPrice);
```
Comment (zoyatanweer) [inline src/pages/cartPage/cartPage.jsx]: remove console.log

**Review by zoyatanweer — APPROVED** [review]
- code is clean and readable
- great work
- good to merge

**Pull request merged**